src/core/parser.js

```diff
@@ -507,8 +507,12 @@ var Parser = (function ParserClosure() {
             stream.pos += i - ENDSTREAM_SIGNATURE_LENGTH;
             break;
           }
-          skipped += scanLength;
-          stream.pos += scanLength;
+          // Since the stream is scanned in `SCAN_BLOCK_SIZE` sized chunks,
+          // it's thus possible that an 'endstream' might be split between
+          // two contiguous `scanBytes` chunks. Avoid skipping over a valid
+          // 'endstream' by subtracting `j` here (fixes issue1536.pdf).
+          skipped += scanLength - j;
+          stream.pos += scanLength - j;
```

Comment:
We already doing subtraction at line 486 above. Which shall properly protect from the case you described in the comment above. Interesting what's missing something the above?